### ISSUE TITLE
DiatonicInterval('major', 1) throws confusing exception

### ISSUE THREAD

**Issue by mallove** [cuthbertLab/music21#506]
The below snippet errors with `{ValueError}2 is not in list`, but it would be more helpful if it read something like, e.g., *there are no major or minor unisons, only __perfect__ unisons*.

```python
music21.note.Note('C').transpose(interval.DiatonicInterval('major', 1))
```

**Comment by mscuthbert** [comment]
Thanks -- great suggestion.

it's always helpful for me to get the simplest version that causes the error.  In this case, it's just  `interval.DiatonicInterval('major', 1)`

**Comment by jacobtylerwalls** [comment]
fixed in fd7d93c

**Comment by mscuthbert** [comment]
Of course it's not true that there are only perfect unisons.  There are augmented etc.
